optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoAnnotationWrappers.java

```diff
@@ -0,0 +1,690 @@
+            }
+            annotationValues = new HashMap<>();
+            if (AllOptaPlannerAnnotationEnum.isOptaPlannerAnnotation(annotationType)) {
+                annotationInstance.valuesWithDefaults(indexView).forEach((value) -> {
+                    switch (value.kind()) {
+                        case BYTE:
+                            annotationValues.put(value.name(),
+                                    value.asByte());
+                            break;
+                        case SHORT:
+                            annotationValues.put(value.name(),
+                                    value.asShort());
+                            break;
+                        case INTEGER:
+                            annotationValues.put(value.name(),
+                                    value.asInt());
+                            break;
+                        case CHARACTER:
+                            annotationValues.put(value.name(),
+                                    value.asChar());
+                            break;
+                        case FLOAT:
+                            annotationValues.put(value.name(),
+                                    value.asFloat());
+                            break;
+                        case DOUBLE:
+                            annotationValues.put(value.name(),
+                                    value.asDouble());
+                            break;
+                        case LONG:
+                            annotationValues.put(value.name(),
+                                    value.asLong());
+                            break;
+                        case BOOLEAN:
+                            annotationValues.put(value.name(),
+                                    value.asBoolean());
+                            break;
+                        case CLASS:
+                            try {
+                                annotationValues.put(value.name(),
+                                        Class.forName(value.asClass().toString()));
+                            } catch (ClassNotFoundException e) {
+                                throw new IllegalStateException(e);
+                            }
+                            break;
+                        case STRING:
+                            annotationValues.put(value.name(),
+                                    value.asString());
+                            break;
+                        case ENUM:
+                            annotationValues.put(value.name(),
+                                    value.asEnum());
+                            break;
+                        case ARRAY:
+                            switch (value.componentKind()) {
+                                case BYTE:
+                                    annotationValues.put(value.name(),
+                                            value.asByteArray());
+                                    break;
```

Comment:
Smells like code duplication with the lines above.

What about an array of arrays - is that unit tested?

Reply:
According to Jandex javadocs, arrays of arrays do not exists as an annotation value:

An annotation value represents a specific name and value combination in the parameter list of an annotation instance. It also can represent a nested array element in the case of an array value.

An annotation value can be any Java primitive:
- byte
- short
- int
- char
- float
- double
- long
- boolean
- 
As well as any the following specialty types:
- String
- Class
- Enum
- Nested annotation
- 
In addition a value can be a *single-dimension array* of any of the above types
To access a value, the proper typed method must be used that matches the expected type of the annotation parameter. In addition, some methods will allow conversion of different types. For example, a byte can be returned as an integer using asInt(). Also all value types support a String representation.
To determine the underlying type, kind() can be used.

Reply:
(also cannot remove the duplication; different methods are used for arrays/single)